common/src/warp_runner/ui_adapter/mod.rs

```diff
@@ -154,17 +154,15 @@ pub async fn dids_to_identity(
 pub async fn fetch_messages_from_chat(
     conv_id: Uuid,
     messaging: &mut super::Messaging,
+    options: MessageOptions,
```

Comment:
It may be confusing to have a function that accepts a `MessageOptions` which lets the caller specify a range, and then override that range (on line 165). It may be better to either calculate the `MessageOptions` before calling `fetch_messages_from_chat` (and remove the `to_take` argument, or make a new function to fetch pinned messages. 

You probably want a new function to fetch all pinned messages because normal messages will be paged and pinned messages will probably be fetched all at once.

Reply:
i split it now into a separate function `fetch_pinned_messages_from_chat` (in case this is needed in future) and rolled back `fetch_messages_from_chat` to how it is in dev